plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2393,6 +2486,19 @@ private HiveColumnHandle createProjectedColumnHandle(HiveColumnHandle column, Li
                 false));
     }
 
+    @Override
+    public List<ColumnHandle> getWriteRedistributionColumns(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) tableHandle;
+        Map<String, HiveColumnHandle> columnNames = new HashMap<>();
+        hiveTableHandle.getPartitionColumns().stream().forEach(column -> columnNames.put(column.getBaseColumnName(), column));
+        hiveTableHandle.getBucketHandle().ifPresent(handle -> handle.getColumns().forEach(column -> columnNames.put(column.getBaseColumnName(), column)));
```

Comment:
I think this is wrong. We want rows for the same bucket to land on the same machine. This is redistributing on the raw bucket column values, rather than on the bucket number (which is a function based on a hash of the bucket values and the bucket count).